### ISSUE TITLE
WFS always same color

### ISSUE THREAD

**Issue by zoran995** [TerriaJS/terriajs#4682]
All WFS layer features will get the same color. For example, all polygons have #93aa00 as `fillColor`. 
This is happening because `GeoJsonCatalogItem`, when generated from WFS, will have an undefined name so `defaultColor` won't do proper randomization of colors. Much better UX would be that features from different WFS layers get a different color.

**Comment by rowanwins** [comment]
Hey @zoran995 

Just to be clear, you'd expect different WFS layers to have different colours, not that you'd expect each feature in a single WFS layer to have different colours?

**Comment by zoran995** [comment]
Yes, different WFS layers to have different colors. Here is an image of the two layers loaded as geojson:
![roads_streams](https://user-images.githubusercontent.com/19208948/91934892-bd43de00-ecec-11ea-88c8-f26adbf17f9b.png)
They got different color because the name of geojson layer is defined. 

p.s. just noticed that this is not only related to WFS, GeoRSS also has the same issue

**Comment by rowanwins** [comment]
Awesome, thanks for confirming @zoran995

**Comment by zoran995** [comment]
Just did some testing for this, we could use a `sourceReference` prop to get name of the parent item.
Something like this: 
```
  @computed get name() {
    if (CatalogMemberMixin.isMixedInto(this.sourceReference)) {
      return super.name || this.sourceReference.name;
    }
    return super.name;
  }
```
inside the `GeoJsonCatalogItem`. Also, the WFS need to send `sourceReference` to `GeoJsonCatalogItem`

**Comment by nf-s** [comment]
Closed by #4701
